erts/preloaded/src/erlang.erl

```diff
@@ -1371,15 +1537,52 @@ list_to_float(_String) ->
 %% list_to_integer/1
 -spec list_to_integer(String) -> integer() when
       String :: string().
-list_to_integer(_String) ->
-    erlang:nif_error(undefined).
+list_to_integer(String) ->
+    Base = 10,
+    case erts_internal:list_to_integer(String, Base) of
+        {Int,[]} ->
```

Comment:
I wonder if it would make sense to just return the integer here to avoid the overhead of the unconditional allocation. The `string` function could wrap the result in a tuple, so we only "pay" for the allocation where necessary.
Shouldn't be a big impact, but for such low-level functions IMO every little thing matters

Reply:
This is not a change I want to make now, in case someone is using the BIF directly despite it being undocumented.